src/aat/java/uk/gov/hmcts/ccd/datastore/befta/DataStoreTestAutomationAdapter.java

```diff
@@ -25,13 +26,32 @@
 
     @Before("@elasticsearch")
     public void skipElasticSearchTestsIfNotEnabled() {
-        if (!ofNullable(System.getenv("ELASTIC_SEARCH_FTA_ENABLED")).map(Boolean::valueOf).orElse(false)) {
+        if (!elasticSearchEnabled()) {
             throw new AssumptionViolatedException("Elastic Search not Enabled");
         }
     }
 
+    @Before
+    public void createUID(Scenario scenario) {
+        String tag = getDataFileTag(scenario);
+        String uid = tag + UUID.randomUUID().toString();
+        uniqueStringsPerTestData.put(tag,uid);
+    }
+
+    private synchronized String getDataFileTag(Scenario scenario) {
+        return scenario.getSourceTagNames().stream()
+            .filter(t -> t.startsWith("@S-"))
+            .findFirst()
+            .map(t -> t.substring(1))
+            .map(Object::toString)
+            .orElse("error cant find tag");
+    }
+
     @Override
     public void doLoadTestData() {
+        if (elasticSearchEnabled()) {
+            new ElasticSearchUtils().deleteIndexesIfPresent();
```

Comment:
let's have a chat about pros and cons of deleting indexes. My fear is that AAT should represent production, and in Production we don't always delete indexes when releasing changes. So if on AAT we always do, we risk of not catching potential issues on AAT but then seeing them on Prod...

Reply:
ok

Reply:
spoke to Ashely, we'll keep this for now as a tactical solution, but we'll start working on the strategic one of using "contains" to verify results